proxy/envoy/config.go

```diff
@@ -711,15 +717,14 @@ func buildInboundListeners(mesh *proxyconfig.MeshConfig, sidecar proxy.Node,
 				listener.Filters = append([]*NetworkFilter{filter}, listener.Filters...)
 			}
 
-			listeners = append(listeners, listener)
-
 		default:
 			glog.V(4).Infof("Unsupported inbound protocol %v for port %#v", protocol, servicePort)
 		}
-	}
 
-	for _, listener := range listeners {
-		applyInboundAuth(listener, mesh)
```

Comment:
just pass the endpoint to the applyInboundAuth function and do the checks there

Reply:
Please do this though. You don't need to restructure or create new functions. Just pass the endpoint object to the applyInboundAuth and then take the decision/skip auth there.

Reply:
I'm not sure if I understand your suggestion correctly. The reason I split the applyInboundAuth function is to reuse shouldApplyAuth in policy.go (for clusters). Moving the call to this applyInboundAuth from the separate "for loop" to the main loop is to make it easier to pass endpoint.